modules/programs/xplr.nix

```diff
@@ -1,30 +1,47 @@
 { config, lib, pkgs, ... }:
 let
   inherit (lib)
-    concatStringsSep types mkIf mkOption mkEnableOption mkPackageOption
-    literalExpression;
+    types mkIf mkOption mkEnableOption mkPackageOption literalExpression;
 
   cfg = config.programs.xplr;
 
   initialConfig = ''
     version = '${cfg.package.version}'
   '';
 
-  # We provide a default version line within the configuration file, which is
-  # obtained from the package's attributes. Merge the initial configFile, a
-  # mapped list of plugins and then the user defined configuration to obtain the
-  # final configuration.
-  pluginPath = if cfg.plugins != [ ] then
-    (''
-      package.path=
-    '' + (concatStringsSep " ..\n"
-      (map (p: ''"${p}/init.lua;${p}/?.lua;"'') cfg.plugins)) + ''
-         ..
-        package.path
-      '')
+  # If `value` is a Nix store path, create the symlink `/nix/store/newhash/${name}/*` 
+  # to `/nix/store/oldhash/*` and returns `/nix/store/newhash`.
+  wrapPlugin = name: value:
+    if lib.isStorePath value then
+      pkgs.symlinkJoin {
+        name = name;
+        paths = [ value ];
+        postBuild = ''
+          mkdir ${name}
+          mv $out/* ${name}/
+          mv ${name} $out/
+        '';
+      }
+    else
+      builtins.dirOf value;
+
+  makePluginSearchPath = p: "${p}/?/init.lua;${p}/?.lua";
```

Comment:
This whole wrapping thing seems strange, what happens if you wrap a plugin in a store path that doesn't have init.lua? I assume it won't match because it'd be `${p}/?/?.lua` not just `${p}/?.lua`

Alternatively, we only allow plugins with `init.lua` (and we can get rid of all this wrapping and wildcard fluff).

Reply:
With only ${p}/?.lua, we'd need to import the plugin as `name.init`, assuming the entrypoint is `init.lua`. It's a lua convention to add init.lua in package path to make the lib easy to import.

Reply:
Then I think my original comment applies?

> what happens if you wrap a plugin in a store path that doesn't have init.lua? I assume it won't match because it'd have to be `${p}/?/?.lua` not just `${p}/?.lua

Reply:
The wrapping does this:
/nix/store/hash/init.lua ->
/nix/store/hash/name/init.lua

So we can add add `package.path = "/nix/store/hash/?/.init.lua;/nix/store/hash/*.lua"` and `require("name")`.

To understand the issue better, let's reproduce it.

1. Create /fakenixstore
2. Git clone sayanarijit/trash-cli.xplr as /fakenixstore/hashsha123 such that sayanarijit/trash-cli.xplr/init.lua is now /fakenixstore/hashsha123/init.lua
3. Get into lua repl (in any dir) and modify package.path in a way so that you can run `require("trash-cli").setup()`.

Here you'll see that the only way to make it work is to:
1. move or symlink `/fakenixstore/hashsha123/*` as `/fakenixstore/hashsha123/trash-cli/*`.
2. In lua repl, run `package.path = "/fakenixstore/hashsha123/?/.init.lua;/nix/fakenixstorestore/hashsha123/*.lua"`

Reply:
@eclairevoyant Based on the most recent message, is it clear why the wrapping method is being used? Do you see any better way? If I or someone can provide more clarification, just say the word. Otherwise, feel free to resolve the conversation.